builders/get-started/eth-compare/security.md

```diff
@@ -0,0 +1,108 @@
+- Blacklisting target contracts (including precompiles) in the function that executes arbitrary call data is **not** considered safe, as other precompiles might be added in the future. Providing whitelisted target contracts in the function that executes the arbitrary call data is considered safe, assuming that the contracts being called are not precompiles, or that in the case they are, the contract making the call does not hold the native token or any XC-20
+
+In the following sections, you'll learn about each of these security considerations through examples.
+
+### Precompiles Can Override a Set Value {: #setting-a-value }
+
+On Ethereum, a smart contract that allows for arbitrary code execution could force the value of a call to be a specific amount (for example, `{value: 0}`), guaranteeing that only that amount of native currency would be sent with the transaction. Whereas on Moonbeam, setting the value of an arbitrary call can be overridden by calling the [native ERC-20 precompile contract](builders/pallets-precompiles/precompiles/erc20){target=_blank} and transferring the native asset through the Substrate API. Since ERC-20s and XC-20s are not native assets, setting the value attribute doesn't provide any protection for these types of assets on Ethereum or Moonbeam.
+
+For example, if you have a contract that allows arbitrary code execution and you pass it encoded call data that transfers the balance of a contract to another address, you could essentially drain the given contract of it's balance.
+
+To get the encoded call data, you can use any of the [ABI encoding functions outlined in the Solidity docs](https://docs.soliditylang.org/en/latest/units-and-global-variables.html#abi-encoding-and-decoding-functions){target=_blank}, including `abi.encodeWithSelector` as seen in the following function:
+
+```
+function getBytes(address _erc20Contract, address _arbitraryCallContract, address _to) public view returns (bytes memory) {
+    // Load ERC-20 interface of contract
+    IERC20 erc20 = IERC20(_erc20Contract);
+    // Get amount to transfer
+    uint256 amount = erc20.balanceOf(_arbitraryCallContract);
+    // Build the encoded call data
+    return abi.encodeWithSelector(IERC20.transfer.selector, _to, amount);
+}
+```
+
+Once you have the encoded call data, you could make an arbitrary call to the [native ERC-20 precompile contract](builders/pallets-precompiles/precompiles/erc20){target=_blank}, set the value of the call to `0`, and pass in the call data in bytes:
+
+```
+function makeArbitraryCall(address _target, bytes calldata _bytes) public {
+    // Value: 0 does not protect against native ERC-20 precompile calls or XCM precompiles
+    (bool success,) = _target.call{value: 0}(_bytes);
+    require(success);
+}
+```
+
+The value of `0` will be overridden by the amount to be transferred as specified in the encoded call data, which in this example is the balance of the contract.
```

Comment:
It might be worth mentioning that we have a way to use our native currency as an ERC20, whereas there is no such thing in Ethereum. In other words:

* Ethereum provides only `value` as a way to transfer funds
* Moonbeam provides `value` and an ERC20-compatible precompile to move the native token

I think this is the fundamental reason this potential security issue arises.